### PR TITLE
Create SettingsFragment

### PR DESCRIPTION
**Pull request by domdomegg**
Part of #455

Next is SingleUploadFragment spinner & Utils, and WelcomeActivity. Might find a way to not require a context in Utils, we'll see.

Tested in API 25, 23, 21, 15 (emulators), and API 24 (Samsung Galaxy S6). I noticed the checkboxes are a [bit weird styled](http://imgur.com/a/6tW47) in API 15 but it is present in the current version so this is not a change I have introduced.

EDIT - Codacy always hates me - I've only added 1 of the 4 'issues', which is a line that is 101 characters instead of 100 that can't really be avoided sensibly.

### CODE REVIEW

**Comment by misaochan** [comment]
I'm guessing the issue with the checkboxes might have been introduced during the day theme/night mode style changes? I don't recall them looking that way when I was testing API 15 back in version 1 a few months ago. 

We should probably make a separate issue for that - it doesn't look too bad, but it probably would look better if it were white.

File: app/src/main/res/xml/preferences.xml [hunk 1]
```diff
@@ -1,29 +1,33 @@
...
-    </Preference>
+        />
 
     <CheckBoxPreference
         android:title="@string/preference_theme"
         android:defaultValue="true"
         android:summary="@string/preference_theme_summary"
         android:key="theme" />
 
+    <Preference android:title="@string/become_a_tester_title"
```
Comment (veyndan) [inline app/src/main/res/xml/preferences.xml]: I don't think modifying the order of the preferences is related to this PR. As a side note, some of the code styling changes in this file can make it hard to review the PR.

**Comment by misaochan** [comment]
We all good to merge?

**Comment by domdomegg** [comment]
I think it should be if @veyndan is happy with it

File: app/src/main/java/fr/free/nrw/commons/settings/Prefs.java [hunk 2]
```diff
@@ -7,8 +7,8 @@
     public static final String DEFAULT_LICENSE = "defaultLicense";
 
     public static class Licenses {
-        public static final String CC_BY_SA_3 = "CC BY-SA";
-        public static final String CC_BY_3 = "CC BY";
+        public static final String CC_BY_SA_3 = "CC BY-SA 3.0";
+        public static final String CC_BY_3 = "CC BY 3.0";
```
Comment (veyndan) [inline app/src/main/java/fr/free/nrw/commons/settings/Prefs.java]: What's the reason for changing the names of these two strings?

Comment (misaochan) [inline app/src/main/java/fr/free/nrw/commons/settings/Prefs.java]: I think the value of those strings should be changed to be more explicit - it was likely just "CC BY" in the legacy app because 4.0 wasn't out yet, but now that it is, it should be renamed to "CC BY 3.0" etc. Should probably be in a different PR though, heh.

Comment (veyndan) [inline app/src/main/java/fr/free/nrw/commons/settings/Prefs.java]: That makes sense. This should ultimately be a PR but I'm fine with this being merged. As long as @domdomegg reads this and knows this for next time.

Comment (domdomegg) [inline app/src/main/java/fr/free/nrw/commons/settings/Prefs.java]: Should have made it clearer, I changed it to match strings.xml and therefore arrays.xml otherwise you'd get mismatches on those settings in singleuploadfragment
